### PR TITLE
Add changing the amount of hands on the GUI depending on your body parts

### LINKED ISSUE #1149: Make HandsComponent aware of body parts
So if you lose one of your hands, you can't use it, or if you have more than two hands it shows, etc.

### PR DESCRIPTION
**Pull request by DrSmugleaf**
Closes #1149 

https://streamable.com/ts6ese

### CODE REVIEW

**Comment by DrSmugleaf** [comment]
> The changes to the BodyManager are _probably_ going to cause massive conflicts with GlassEclipse's work. It'd be better if you cut out all the styling changes and kept only the actual added/removed hooks.

Reverted in https://github.com/space-wizards/space-station-14/pull/1406/commits/13e92db3902eb80f1fc16bfc754db2a0fd606ac9

File: Content.Server/Health/BodySystem/BodyManagerComponent.cs [hunk 1]
```diff
@@ -355,5 +373,24 @@ public IEntity DropBodyPart(BodyPart part)
...
+
+        private void BodyPartRemoved(BodyPart part, string slotName)
+        {
+            var args = new BodyPartRemovedEventArgs(part, slotName);
+
+            foreach (var component in Owner.GetAllComponents<IBodyPartRemoved>())
+            {
+                component.BodyPartRemoved(args);
+            }
+        }
```
Comment (Zumorica) [inline Content.Server/Health/BodySystem/BodyManagerComponent.cs:394]: Couldn't this all simply be two events?

File: Content.Shared/GameObjects/Components/Items/SharedHandsComponent.cs [hunk 2]
```diff
@@ -1,24 +1,43 @@
...
+            Location = location;
+        }
+    }
+
     // The IDs of the items get synced over the network.
     [Serializable, NetSerializable]
     public class HandsComponentState : ComponentState
     {
-        public readonly Dictionary<string, EntityUid> Hands;
+        public readonly List<SharedHand> Hands;
```
Comment (Zumorica) [inline Content.Shared/GameObjects/Components/Items/SharedHandsComponent.cs:37]: Make this an array

File: Content.Server/Health/BodySystem/BodyManagerComponent.cs (hunk 1 above)
Comment (DrSmugleaf) [reply Content.Server/Health/BodySystem/BodyManagerComponent.cs:394]: Yes but BodySystem loads the body parts in ExposeData right now, making it very hard to hook into if it was events.

**Pull request merged**